march_hardware/src/IMotionCube.cpp

```diff
@@ -304,7 +304,7 @@ float IMotionCube::getIMCVoltage()
 
 float IMotionCube::getMotorVoltage()
 {
-  return this->read16(this->miso_byte_offsets_.at(IMCObjectName::MotorVoltage)).ui;
+  return this->read16(this->miso_byte_offsets_.at(IMCObjectName::MotorVoltage)).f;
```

Comment:
As Rutger already mentioned, this is not possible since float can only be a 32 bit value. You would have to do some manual conversions. You should convert this back for now and we'll see later about the conversion.
```suggestion
  return this->read16(this->miso_byte_offsets_.at(IMCObjectName::MotorVoltage)).ui;
```